lib/MemoryClient.js

```diff
@@ -1,51 +1,58 @@
+let _Promise
 export default class MemoryClient {
   constructor (options) {
     this.options = options
     this.data = {}
+    _Promise = options.promiseDependency || global.Promise
```

Comment:
why not `this._Promise` or at least wrap it around `if(_Promise)`

Reply:
Will do, any preference?

Reply:
Let's just wrap it.

Reply:
I used `this.Promise as it seemed to make more sense`